app/models/report.rb

```diff
@@ -14,4 +19,30 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def create_with_mentions
+    Report.transaction do
+      content.include?('http://localhost:3000') ? save && create_mentions : save
+    end
+  end
```

Comment:
メンションがない時にcreate_mentionsが動いても問題ないように作られていれば（もしかしたら現状でも大丈夫かも？）、下記のようにかけてスマートでわかりやすいかもです。

```suggestion
  def create_with_mentions
   transaction do
      save && create_mentions
    end
  end
```

贅沢を言えば、create_mentions, update_mensionsをsave_mentionsと言う名前で共通化できたら

`save && save_mentions`と書けるのでより読む人にとって理解しやすいかもです〜。

Reply:
なるほどです！！こういった書きかたがあるんですね！！
勉強になります🙇‍♂️

Reply:
@kurumadaisuke ちょっと僕は動かしていないので実際に下記の場合に問題なく動くのかわかってないのですが大丈夫な感じでしょうか？

> メンションがない時にcreate_mentionsが動いても問題ないように作られていれば

Reply:
このところが動作確認を行った結果問題ありませんでした🙏
メンションがない日報作成の場合、ReportMention.allの結果は以下のようになっておりました🙏
```
irb(main):001:0> ReportMention.all
  ReportMention Load (0.6ms)  SELECT "report_mentions".* FROM "report_mentions"
=> [] 
```